ironfish/src/workerPool/tasks/decryptNotes.ts

```diff
@@ -237,7 +239,7 @@ export class DecryptNotesTask extends WorkerTask {
           forSpender: false,
           hash: note.hash(),
           nullifier:
-            currentNoteIndex !== null
+            currentNoteIndex !== null && spendingKey
```

Comment:
Passing null here means the note being decrypted is not in the tree. Now if this is will all notes decrypted will be interpreted as not being on the tree. Do we need to fix this first in another PR? This seems like it could hint that we may be dead in the water with this. What's the implication to the wallet to not being able to create nullifiers to notes?

Reply:
The downstream of this is here:
https://github.com/iron-fish/ironfish/blob/ece1454b6b355146bd137edcf6dc0b55edd3af14/ironfish/src/wallet/walletdb/walletdb.ts#L574-L580
Particularly for the `nullifierToNoteHash`. 
I will look at where that table is utilized

Reply:
It seems like the main issue is this line:
https://github.com/iron-fish/ironfish/blob/1a051809dfd6f59d7178e174a5a79d98bf5fc756/ironfish/src/wallet/walletdb/walletdb.ts#L505-L510

We are getting note hashes by their nullifiers. I think you are right, the nullifierToNoteHash is used in a variety of very important methods for handling `spends`. The assumption is that the note is only a spend if the table has a corresponding `nullifierToNoteHash`. I will convene with @hughy and we can discuss plan of action.